### PR TITLE
Filtering solved/unsolved, Q&A format for tags

### PR DESCRIPTION
**Pull request by imorland**
image
image
image

### CODE REVIEW

File: js/src/forum/index.js [hunk 1]
```diff
@@ -40,4 +48,93 @@ app.initializers.add('fof/best-answer', () => {
             label: app.translator.trans('fof-best-answer.forum.notification.preferences.best_answer_in_discussion'),
         });
     });
+
+    extend(IndexPage.prototype, 'sidebarItems', function (items) {
+        const tag = this.currentTag();
+
+        if (!tag || !tag.isQnA()) return;
```
Comment (davwheat) [inline js/src/forum/index.js:55]: We can use fancy new [optional chaining](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining) operator for this.
```suggestion
        if (!tag?.isQnA?.()) return;
```

File: js/src/forum/index.js [hunk 2]
```diff
@@ -40,4 +48,93 @@ app.initializers.add('fof/best-answer', () => {
...
+            canStartDiscussion ? 'fof-best-answer.forum.index.ask_question' : 'fof-best-answer.index.cannot_ask_question'
+        );
+
+        items.replace('startDiscussion', cta);
+    });
+
+    extend(IndexPage.prototype, 'viewItems', function (items) {
+        const tag = this.currentTag();
+
+        if (!tag || !tag.isQnA()) {
```
Comment (davwheat) [inline js/src/forum/index.js:69]:
```suggestion
        if (!tag?.isQnA?.()) {
```

File: js/src/forum/index.js [hunk 3]
```diff
@@ -40,4 +48,93 @@ app.initializers.add('fof/best-answer', () => {
...
+        const options = ['all', 'solved', 'unsolved'];
+
+        const selected = app.discussions.bestAnswer;
+
+        items.add(
+            'solved-filter',
+            Dropdown.component({
+                buttonClassName: 'Button',
+                label: app.translator.trans(`fof-best-answer.forum.filter.${(options[selected] || Object.keys(options).map((key) => options[key])[0])}_label`),
+                accessibleToggleLabel: 'assessible label',
```
Comment (davwheat) [inline js/src/forum/index.js:87]: :(

Comment (imorland) [reply js/src/forum/index.js:87]: Whoops 🙈

**Comment by Ralkage** [comment]
Filtering solved/unsolved works and the refactored options relocated to a Tag level works as well!

Additionally, `best-answer:unsolved` and `best-answer:solved` search filters both return the expected results; really loving these changes!! We're good to merge @imorland 👍

**Review by Ralkage — APPROVED** [review]

**Review by davwheat — APPROVED** [review]
Whoops, I thought I approved this already!

Fantastic job as always Ian :)

**Pull request merged**